### PR TITLE
controllers: adds $schema to payload

### PR DESCRIPTION
**Pull request by pamfilos**
Signed-off-by: Pamfilos Fokianos <pamfilosf@gmail.com>

### CODE REVIEW

File: src/invenio-records-js/controllers/InvenioRecordsCtrl.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ function InvenioRecordsCtrl($scope, $rootScope, $q, $window, $location,
             url: vm.invenioRecordsEndpoints[type],
             method: (method || 'PUT').toUpperCase(),
             data: {
+              '$schema': vm.invenioRecordsEndpoints.schema,
```
Comment (jirikuncar) [inline src/invenio-records-js/controllers/InvenioRecordsCtrl.js]: Shouldn't this be a part of `_data` (`metadata`)? @lnielsen @drjova

Comment (pamfilos) [inline src/invenio-records-js/controllers/InvenioRecordsCtrl.js]: No.. For it to work now it should be on the same level as ```metadata```, not inside

**Comment by lnielsen** [comment]
We need to refactor how the JSON payload is provided. E.g. adding ``$schema`` will not work for the Zenodo REST API. I think the best is that we can provide a callback which can "massage" the payload before sending it.

@drjova Any ideas for how best to solve it?

**Comment by lnielsen** [comment]
Additionally, since we're soon to release Zenodo, we probably need two branches (maint/master, master/next), so that we don't block each other.

**Comment by drjova** [comment]
Superseded by  https://github.com/inveniosoftware/invenio-records-js/pull/24/commits/f0c65e254dd2a345c93388b828d98ef9f448c54c you can now add additional data to the payload by using the existing ``extra-params`` attribute

```html
<invenio-records
  extra-params='{"headers": {"foo": "bar"}, "data": {"$schema": "foofoo"}}'
>
</invenio-records>
```
With that fix you it'll work exactly as it is on ``invenio-search-js`` any Invenio app can have its own custom ``data``, ``headers`` etc. :)

**Comment by lnielsen** [comment]
@drjova But I think the problem is as well that it's adding the ``{'metadata': ..}`` key…..which is needed on zenodo but not analysis preservation

**Comment by drjova** [comment]
@lnielsen hmm we have it already there https://github.com/inveniosoftware/invenio-records-js/blob/master/src/invenio-records-js/controllers/InvenioRecordsCtrl.js#L187-L189 I'll make it configurable

**Comment by drjova** [comment]
superseded by https://github.com/inveniosoftware/invenio-records-js/pull/24

**Pull request closed without merging**